File: Flutter_UI/lib/shared/models/order_enums.dart
// lib/shared/models/order_enums.dart

/// Matches backend OrderInputType enum
enum OrderInputType {
  image(0),    // For both File Upload and Camera
  voice(1),    // For Voice Recording
  text(2);     // For Text and WhatsApp

  final int value;
  const OrderInputType(this.value);

  static OrderInputType fromValue(int value) {
    return OrderInputType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => OrderInputType.image,
    );
  }
}

/// Matches backend OrderType enum
enum OrderType {
  notSet(0),
  otc(1),                    // Over-the-Counter
  prescriptionDrugs(2);      // Requires prescription

  final int value;
  const OrderType(this.value);

  String get displayName {
    switch (this) {
      case OrderType.otc:
        return 'Over-the-Counter (OTC)';
      case OrderType.prescriptionDrugs:
        return 'Prescription Required';
      default:
        return 'Not Set';
    }
  }

  static OrderType fromValue(int value) {
    return OrderType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => OrderType.notSet,
    );
  }
}

/// Order status enum
enum OrderStatus {
  pendingPayment(0),
  assignedToChemist(1),
  rejectedByChemist(2),
  acceptedByChemist(3),
  billUploaded(4),
  paid(5),
  outForDelivery(6),
  completed(7);

  final int value;
  const OrderStatus(this.value);

  String get displayName {
    switch (this) {
      case OrderStatus.pendingPayment:
        return 'Pending Payment';
      case OrderStatus.paid:
        return 'Paid';
      case OrderStatus.assignedToChemist:
        return 'Assigned to Chemist';
      case OrderStatus.rejectedByChemist:
        return 'Rejected by Chemist';
      case OrderStatus.acceptedByChemist:
        return 'Accepted by Chemist';
      case OrderStatus.billUploaded:
        return 'Bill Uploaded';
      case OrderStatus.outForDelivery:
        return 'Out for Delivery';
      case OrderStatus.completed:
        return 'Completed';
    }
  }

  static OrderStatus fromValue(int value) {
    return OrderStatus.values.firstWhere(
      (status) => status.value == value,
      orElse: () => OrderStatus.pendingPayment,
    );
  }
}

enum DeliveryType {
  homeDelivery,  // 0
  storePickup,   // 1
}

enum OrderUrgency {
  regular,  // 0
  urgent,   // 1
}

File: WebApp/src/models/OrderEnums.ts
export enum OrderType {
  NotSet = 0,
  OTC = 1,
  PrescriptionDrugs = 2,
}

export const OrderTypeLabel: Record<OrderType, string> = {
  [OrderType.NotSet]: 'Not Set',
  [OrderType.OTC]: 'Over-the-Counter (OTC)',
  [OrderType.PrescriptionDrugs]: 'Prescription Required',
};

export enum OrderInputType {
  Image = 0,
  Voice = 1,
  Text = 2,
}

export const OrderInputTypeLabel: Record<OrderInputType, string> = {
  [OrderInputType.Image]: 'Image',
  [OrderInputType.Voice]: 'Voice',
  [OrderInputType.Text]: 'Text',
};

export enum OrderStatus {
  PendingPayment = 0,
  AssignedToChemist = 1,
  RejectedByChemist = 2,
  AcceptedByChemist = 3,
  BillUploaded = 4,
  Paid = 5,
  OutForDelivery = 6,
  Completed = 7,
  AssignedToCustomerSupport = 8,
}

export const OrderStatusLabel: Record<number, string> = {
  0: 'Pending Payment',
  1: 'Assigned to Chemist',
  2: 'Rejected by Chemist',
  3: 'Accepted by Chemist',
  4: 'Bill Uploaded',
  5: 'Paid',
  6: 'Out for Delivery',
  7: 'Completed',
  8: 'Assigned to Customer Support',
};

export enum OrderPaymentStatus {
  NotPaid = 0,
  PartiallyPaid = 1,
  FullyPaid = 2,
}

export enum AssignmentStatus {
  Assigned = 0,
  Accepted = 1,
  Rejected = 2,
}

export enum AssignTo {
  Customer = 0,
  Chemist = 1,
  CustomerSupport = 2,
  Delivery = 3,
}

export enum AssignedByType {
  System = 0,
  CustomerSupport = 1,
}

export enum RegionType {
  CustomerSupport = 0,
  DeliveryBoy = 1,
}

export type UserRole = 'Admin' | 'Chemist' | 'CustomerSupport' | 'Manager' | 'DeliveryBoy' | 'Customer';
